### ISSUE TITLE
key-event.js is not working in IE10

### ISSUE THREAD

**Issue by ernsheong** [ccampbell/mousetrap#279]
`KeyEvent.prototype.fire` returns "Error: Invalid argument." in IE10.

Any help with this is appreciated.

**Comment by lepolt** [comment]
Just ran into this issue myself... any progress to report?

**Comment by ccampbell** [comment]
`key-event.js` was something I made to assist with unit testing. It was never designed to work everywhere. If you need to trigger key events in your app you should use `Mousetrap.trigger` instead.

**Comment by lepolt** [comment]
I'm using `key-event.js` in my unit tests as well :) We're doing integration testing and this was the best way to simulate key events in our system.

It's a helpful library, but apparently doesn't play well with IE10...

**Comment by ernsheong** [comment]
For integration testing I would argue that a driver-based tool like Selenium might be worth taking a look at. You can send keys, etc. As for unit tests, you can always fall back to creating the key event yourself, YMMV

**Comment by workmanw** [comment]
@ernsheong I think that's actually good for acceptance testing. I believe @lepolt is referring to testing our application's mousetrap wrapper.

**Comment by ccampbell** [comment]
It sounds like in that case you are actually testing the functionality of Mousetrap itself (making sure key events in the browser trigger things to happen). 

It might make sense to use `Mousetrap.trigger` in your tests instead of simulating key events and just make sure when you call trigger and pass in the right keys the callback you are expecting to fire is fired. 

That would mean that if a broken version of mousetrap was included that your tests could succeed even tho they don't actually work correctly but if mousetrap is broken then that is not the fault of your application. 

All that being said if someone wants to take a stab at getting `key-event.js` to work on old IE I'd be happy to take a look at it/merge it.

**Comment by lepolt** [comment]
We are actually using multiple instances of Mousetrap because in some cases we "stack" dialogs and don't want the Mousetrap instances on the bottom to respond when there is an open dialog. Simulating a browser event allows us to test the un/pause logic that we are using end to end.

**Comment by ernsheong** [comment]
@lepolt You should consider using https://craig.is/killing/mice#wrapping instead of this pause/unpause logic. We have done this successfully in our app, but it depends on your use case.

**Comment by lepolt** [comment]
Yes, we tried that, but it doesn't work with our application framework.
